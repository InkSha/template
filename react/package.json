{
  "name": "demo",
  "version": "0.1.0",
  "dependencies": {
    "@reduxjs/toolkit": "^2.0.1",
    "@testing-library/jest-dom": "^5.17.0",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.5.2",
    "@types/mockjs": "^1.0.10",
    "@types/node": "^16.18.68",
    "@types/react": "^18.2.46",
    "@types/react-dom": "^18.2.18",
    "@types/react-redux": "^7.1.33",
    "@types/react-router-dom": "^5.3.3",
    "antd": "^5.12.5",
    "axios": "^1.6.3",
    "eslint-plugin-prettier": "^5.1.2",
    "history": "4.10.1",
    "mockjs": "^1.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^9.0.4",
    "react-router-dom": "^6.21.1",
    "react-scripts": "5.0.1",
    "typescript": "^4.9.5",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "craco start",
    "build": "craco build",
    "test": "craco test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
    }
  },
  "lint-staged": {
    "*.{tx,tsx,js,jsx}": [
      "eslint --config .eslintrc.js"
    ],
    "**/*": "prettier --write"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
    "@commitlint/cli": "^18.4.3",
    "@commitlint/config-angular": "^18.4.3",
    "@craco/craco": "^7.1.0",
    "@typescript-eslint/eslint-plugin": "^6.16.0",
    "@typescript-eslint/parser": "^6.16.0",
    "eslint": "^8.56.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-react": "^7.33.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "husky": "^8.0.3",
    "lint-staged": "^15.2.0",
    "node-sass": "^9.0.0",
    "prettier": "^3.1.1",
    "prettier-eslint": "^16.2.0",
    "sass-loader": "^13.3.3"
  }
}
