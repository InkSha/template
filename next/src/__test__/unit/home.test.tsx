import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '@/app/page'

describe('测试首页渲染', () => {
  it('渲染 main', () => {
    render(<Page />)
    const main = screen.getByRole('main')
    //   main元素在 文档中
    expect(main).toBeInTheDocument()
  })

  it('测试快照', () => {
    const { container } = render(<Page />)
    expect(container).toMatchSnapshot('home')
  })
})
