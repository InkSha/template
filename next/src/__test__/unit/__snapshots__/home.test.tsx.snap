// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试首页渲染 测试快照: home 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="description"
    >
      <p>
        Get started by editing 
        <code
          class="code"
        >
          src/app/page.tsx
        </code>
      </p>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          en
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="en"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1br1jja"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div>
        <a
          href="https://vercel.com?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          By
           
          <img
            alt="Vercel Logo"
            class="vercelLogo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="24"
            src="/vercel.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
      </div>
    </div>
    <div
      class="center"
    >
      <img
        alt="Next.js Logo"
        class="logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="37"
        src="/next.svg"
        style="color: transparent;"
        width="180"
      />
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="to"
        >
          see
        </span>
        <a
          class="link"
          href="/count"
        >
          redux
        </a>
        <span
          class="see"
        >
          example
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="grid"
    >
      <a
        class="card"
        href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Docs
           
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Find in-depth information about Next.js features and API.
        </p>
      </a>
      <a
        class="card"
        href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Learn
           
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Learn about Next.js in an interactive course with quizzes!
        </p>
      </a>
      <a
        class="card"
        href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Templates
           
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Explore starter templates for Next.js
        </p>
      </a>
      <a
        class="card"
        href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2>
          Deploy
           
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          Instantly deploy your Next.js site to a shareable URL with Vercel.
        </p>
      </a>
    </div>
  </main>
</div>
`;
